components/script/dom/uievent.rs

```diff
@@ -75,15 +75,32 @@ impl UIEvent {
         detail: i32,
         can_gc: CanGc,
     ) -> DomRoot<UIEvent> {
-        let ev = UIEvent::new_uninitialized_with_proto(window, proto, can_gc);
-        ev.InitUIEvent(
-            type_,
-            bool::from(can_bubble),
-            bool::from(cancelable),
-            view,
-            detail,
-        );
-        ev
+        UIEvent::initialize_ui_event(
+            window, proto, type_, can_bubble, cancelable, view, detail, can_gc,
+        )
+    }
+
+    /// based on <https://w3c.github.io/uievents/#initialize-a-uievent>
+    #[allow(clippy::too_many_arguments)]
+    pub fn initialize_ui_event(
```

Comment:
According to spec, this method should take UIEvent itself `&self` and return nothing. See how  [MouseEvent::initialize_mouse_event](https://github.com/servo/servo/blob/main/components/script/dom/mouseevent.rs#L184) does this.

Reply:
Actually I did that initially, but later I tried to bridge `new_with_proto` and make `initialize_ui_event` suitable for use with `new_with_proto`. I will make this method private (for use in `uievent` module only) and create one that follow the spec (for use externally).

Reply:
https://github.com/servo/servo/pull/34524/commits/41e209df4c3cbd7ed067c2fb071562f7903c267e